Test fail in test_db

```
dug_seis/tests/test_db.py:1270: AssertionError
===================================================================================== warnings summary ======================================================================================
../../../.local/mambaforge/envs/dug_seis/lib/python3.11/site-packages/obspy/core/util/base.py:26
  /home/eule/.local/mambaforge/envs/dug_seis/lib/python3.11/site-packages/obspy/core/util/base.py:26: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
    import pkg_resources

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================== short test summary info ==================================================================================
FAILED dug_seis/tests/test_db.py::test_add_origin_to_event_and_change_preferred_origin - assert Origin(resource_id=ResourceIdentifier(id="smi:local/0ca79b16-1486-4be0-90d9-7e253fee7e9d"), time=UTCDateTime(1970, 1, 1, 0, 0, 10), longitude=32.0, latitude=33.034, depth=12020.0) == Origin(resource_id=ResourceIdentifier(id="smi:local/f9e76334-adca-4483-a33f-cb4a45c8766e"), time=UTCDateTime(2024, 5, 23, 7, 1, 57, 640091), longitude=12.0, latitude=32.034, depth=10020.0)
  
  Full diff:
  - Origin(resource_id=ResourceIdentifier(id="smi:local/f9e76334-adca-4483-a33f-cb4a45c8766e"), time=UTCDateTime(2024, 5, 23, 7, 1, 57, 640091), longitude=12.0, latitude=32.034, depth=10020.0)
  + Origin(resource_id=ResourceIdentifier(id="smi:local/0ca79b16-1486-4be0-90d9-7e253fee7e9d"), time=UTCDateTime(1970, 1, 1, 0, 0, 10), longitude=32.0, latitude=33.034, depth=12020.0)
```

It is triggered only in ~50% of test runs.

--- a/dug_seis/db/sqlite_db_backend.py
+++ b/dug_seis/db/sqlite_db_backend.py
@@ -155,6 +155,8 @@ class _SQLiteBackend:
 
         results = []
 
+        sql += " ORDER BY _oid ASC"
+
         for row in self.cursor.execute(sql).fetchall():
             result = {}
             for name, r in zip(column_names, row):
@@ -588,7 +590,8 @@ class _SQLiteBackend:
                 # Origins are stored via origin references.
                 origin_refs = self.cursor.execute(
                     "SELECT _oid, _parent_oid, originID from OriginReference "
-                    f"WHERE _parent_oid in ({','.join(['?'] * len(object_results))})",
+                    f"WHERE _parent_oid in ({','.join(['?'] * len(object_results))}) "
+                    "ORDER BY _oid ASC",
                     [o._object_id for o in object_results],
                 ).fetchall()
                 if origin_refs:

--- a/dug_seis/tests/test_db.py
+++ b/dug_seis/tests/test_db.py
@@ -25,10 +25,6 @@ from dug_seis.db.db import DB
 DATA_DIR = pathlib.Path(__file__).parent / "data"
 
 
-def _sort_origins(event):
-    event.origins = sorted(event.origins, key=lambda x: str(x.resource_id))
-
-
 def test_basic_db(tmp_path):
     url = f"sqlite://{tmp_path / 'db.sqlite'}"
     db = DB(url=url)
@@ -1213,8 +1209,8 @@ def test_write_and_read_same_picks_used_in_multiple_arrivals():
     # unused pick.
     del event.picks[0]
 
-    _sort_origins(event)
-    _sort_origins(events[0])
+    event
+    events[0]
 
     # Now they are the same.
     assert event == events[0]
@@ -1266,8 +1262,8 @@ def test_add_origin_to_event_and_change_preferred_origin():
 
     # Adding to the database should change it.
     db.update_event(event=event)
-    assert _sort_origins(db.get_objects(object_type="Event")[0]) == _sort_origins(event)
-    assert db.get_objects(object_type="Event")[0].preferred_origin() == event.origins[1]
+    assert db.get_objects(object_type="Event")[0] == event
+    assert db.get_objects(object_type="Event")[0].preferred_origin() == event.preferred_origin()
 
     assert db.count("Event") == 1
     assert db.count("Origin") == 2

